src/components/RisChipsInput/RisChipsInput.vue

```diff
@@ -0,0 +1,105 @@
+<script lang="ts" setup>
+import { computed, ref } from "vue";
+import IconCancel from "~icons/ic/baseline-cancel";
+import ChipInput from "./ChipInput.vue";
+
+const props = defineProps<{
+  ariaLabel?: string;
+  hasError?: boolean;
+  mask?: string;
+  placeholder?: string;
+}>();
+
+const model = defineModel<string[]>({ required: true });
+
+const newChipInputRef = ref<HTMLInputElement | null>(null);
+
+const editingChipIndex = ref<number | null>(null);
+const newChipText = ref<string>("");
+
+function toggleChipEditMode(chipIndex: number) {
+  editingChipIndex.value = chipIndex;
+}
+
+function onAddChip(value: string) {
+  if (!value) return;
+
+  model.value = [...model.value, value.trim()];
+  newChipText.value = "";
+}
+
+function onComplete() {
+  model.value = [...model.value].filter(Boolean);
+  editingChipIndex.value = null;
+}
+
+function onDeleteChip(index: number) {
+  model.value = model.value.filter((_, i) => i !== index);
+
+  editingChipIndex.value = null;
+}
+
+function focusNewChipInput() {
+  if (!editingChipIndex.value) {
+    newChipInputRef.value?.focus();
+  }
+}
+
+const conditionalClasses = computed(() => ({
+  "!shadow-red-800 !bg-red-200": props.hasError,
+}));
+</script>
+
+<template>
+  <div
```

Comment:
Maybe we can consider a role="group" or role="listbox" for this main div, with appropriate aria-labelledby or aria-label if it's meant to function as a compound input?